django/contrib/admin/options.py

```diff
@@ -1505,15 +1506,32 @@ def get_changeform_initial_data(self, request):
         Get the initial form data from the request's GET params.
         """
         initial = dict(request.GET.items())
+        formset_data = defaultdict(lambda: defaultdict(defaultdict))
+        formset_query_regex = r'inlinemode_set-(\d+)-(\d+)-(.*)'
         for k in initial:
             try:
+                match_obj = re.match(formset_query_regex, k)
+                if match_obj:
+                    inline_model_index = int(match_obj.group(1))
+                    inline_model_form_index = int(match_obj.group(2))
+                    inline_model_form_fieldname = match_obj.group(3)
+                    model_field = self.inlines[inline_model_index].model._meta.get_field(inline_model_form_fieldname)
+                    if isinstance(model_field, models.ManyToManyField):
+                        formset_data[inline_model_index][inline_model_form_index][inline_model_form_fieldname] \
+                            = initial.get(k).split(",")
+                    else:
+                        formset_data[inline_model_index][inline_model_form_index][inline_model_form_fieldname] \
+                            = initial.get(k)
+
                 f = self.model._meta.get_field(k)
             except FieldDoesNotExist:
                 continue
+            except IndexError:
+                continue
```

Comment:
Is this tested? What if `fieldname` is invalid? That would raise a `KeyError` so that should be handled too?